lib/bigint_pk.rb

```diff
@@ -1,37 +1,23 @@
-require 'active_support/all'
 require 'bigint_pk/version'
+require 'bigint_pk/railtie'
+require 'active_record'
 
 module BigintPk
-  mattr_accessor :enabled
+  extend self
 
-  autoload :Generators, 'generators/bigint_pk'
-
-  def self.setup
-    yield self
-  end
-
-  def self.enabled= value
-    install_patches! if value
+  def enable!
+    install_patches!
   end
 
   private
 
-  def self.install_patches!
-    install_primary_key_patches!
-    install_foreign_key_patches!
-  end
-
-  def self.install_primary_key_patches!
-    ActiveRecord::Base.establish_connection
-    ca = ActiveRecord::ConnectionAdapters
-
-    if ca.const_defined? :PostgreSQLAdapter
-      ca::PostgreSQLAdapter::NATIVE_DATABASE_TYPES[:primary_key] = 'bigserial primary key'
-    end
-
-    if ca.const_defined? :AbstractMysqlAdapter
-      ca::AbstractMysqlAdapter::NATIVE_DATABASE_TYPES[:primary_key] = 'bigint(20) auto_increment PRIMARY KEY'
-      ca::AbstractMysqlAdapter::NATIVE_DATABASE_TYPES[:integer] = { :name => "bigint", :limit => 6 }
+  module DefaultBigintPrimaryKey
+    def primary_key(name, type = :primary_key, **options)
```

Comment:
In Rails 4.2 it used the `NATIVE_DATABASE_TYPES` mapping to get the sql type to be used in the column definition. In Rails 5 this mapping is not used anymore and instead of it the ColumnDefinition object now have a DSL for all the types. So, in order to actually make this gem work with Rails 5 we need to override the primary key column builder to use bigint.